### PR TITLE
Us99897 ta128660 remove nav from iframe

### PR DESCRIPTION
**Pull request by benyamin-noori**
Look for a "message" event from the iframe and if received, send a message event to iframe with elements to be deleted.

### CODE REVIEW

**Comment by jwong-d2l** [comment]
Any chance we could get some tests for this new behaviour?

**Comment by benyamin-noori** [comment]
I'll start working on it.

File: d2l-sequence-viewer.html [hunk 1]
```diff
@@ -313,5 +312,58 @@
...
+			}
+
+			if (typeof(eventData) === 'object' && eventData.eventType && eventData.eventType === 'd2l-iframe-load') {
+				fireRemoveNavElementsEvent();
+			}
+
+		}, false);
+
+		function fireRemoveNavElementsEvent() {
+			var content = document.querySelector('d2l-sequence-viewer').shadowRoot.querySelector('d2l-sequences-content-router').shadowRoot.querySelector('d2l-sequences-content-link').shadowRoot.querySelector('#content');
```
Comment (jbydeley-d2l) [inline d2l-sequence-viewer.html]: I'd like to see this line broken up a bit more. Also this can be a `const`

File: d2l-sequence-viewer.html [hunk 2]
```diff
@@ -313,5 +312,58 @@
...
+
+			// prepare postMessage data.
+			var origin = window.location.protocol + '//' + window.location.host;
```
Comment (jbydeley-d2l) [inline d2l-sequence-viewer.html]:
```suggestion
			const origin = `${window.location.protocol}//${window.location.host}`;
```

File: d2l-sequence-viewer.html [hunk 3]
```diff
@@ -313,5 +312,58 @@
...
+			var eventData = {
```
Comment (jbydeley-d2l) [inline d2l-sequence-viewer.html]:
```suggestion
			const eventData = {
```

File: d2l-sequence-viewer.html [hunk 4]
```diff
@@ -313,5 +312,58 @@
...
+				eventType: 'd2l-element-hider',
+				selectors: selectors
```
Comment (jbydeley-d2l) [inline d2l-sequence-viewer.html]:
```suggestion
				selectors
```

File: d2l-sequence-viewer.html [hunk 5]
```diff
@@ -313,5 +312,58 @@
...
+					innerHTML: 'Exit'
+				}
+			},
+			{
+				cssSelector: 'div.d2l-page-header' // Page header including 'settings' and 'help' buttons.
+			}
+		];
+
+		window.addEventListener('message', function(event) {
+			var eventData;
```
Comment (jbydeley-d2l) [inline d2l-sequence-viewer.html]: You could change this to wrap everything in the try / catch and then you wouldn't need to set `eventData = event.data` in the catch.

```
try {
   eventData = JSON.parse(event.data)
   if( eventData && eventData.eventType && eventData.eventType === 'd2l-iframe-load' ) {
       fireRemoveNavElementsEvent()
   }
} catch(err) {
}
```

**Pull request closed without merging**